### PR TITLE
Add eventing dashboards

### PR DESCRIPTION
**Pull request by berkanercan**
**Description**

With this PR, we introduce the dashboards for nats based eventing.

Changes proposed in this pull request:

- Eventing / Latency Dashboard
- Eventing / Pods Dashboard
- Eventing / Delivery Dashboard

### CODE REVIEW

**Review by sayanh — DISMISSED** [review]

File: resources/eventing/charts/nats/templates/50-servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    {{- include "nats.labels" . | nindent 4 }}
+    prometheus: monitoring
+spec:
+  selector:
+    matchLabels: {{ include "nats.labels.selectorLabels" . | nindent 6 }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace | quote }}
+  endpoints:
+    - port: metrics
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/nats/templates/50-servicemonitor.yaml]: Done

File: resources/eventing/charts/nats/templates/50-servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      interval: 10s
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/nats/templates/50-servicemonitor.yaml]: Done

File: resources/eventing/charts/nats/templates/40-cr.yaml [hunk 3]
```diff
@@ -7,8 +7,13 @@ spec:
   size: {{ .Values.clusterSize }}
   version: "2.1.8"
   pod:
+    enableMetrics: true
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/nats/templates/40-cr.yaml]: Done

File: resources/eventing/charts/nats-controller/templates/peerauthentication.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{{- if eq .Values.global.eventing.backend "nats" }}
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: {{ .Chart.Name }}
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/nats-controller/templates/peerauthentication.yaml]: Done

File: resources/eventing/charts/event-publisher-nats/templates/service.yaml [hunk 5]
```diff
@@ -8,22 +8,24 @@ spec:
...
+      protocol: TCP
       port: 80
-      targetPort: {{ .Values.nats.config.port }}
+      targetPort: {{ .Values.nats.config.portName }}
 ---
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ include "event-publisher-nats.serviceName" . }}{{ .Values.metrics.service.nameSuffix }}
+  name: {{ include "event-publisher-nats.serviceName" . }}-metrics
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/event-publisher-nats/templates/service.yaml]: Restored

File: resources/eventing/charts/event-publisher-nats/templates/peerauthentication.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{{- if eq .Values.global.eventing.backend "nats" }}
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: {{ .Chart.Name }}
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/event-publisher-nats/templates/peerauthentication.yaml]: Done

File: resources/eventing/charts/nats-controller/values.yaml [hunk 7]
```diff
@@ -17,8 +17,10 @@ image:
 podSecurityContext:
 
 metrics:
-  service:
-    nameSuffix: "-metrics"
+  config:
+    port: 8080
+    portName: http-metrics
+    interval: 10s
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/nats-controller/values.yaml]: Done

File: resources/eventing/charts/dashboards/templates/delivery.yaml [hunk 8]
```diff
@@ -0,0 +1,621 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "dashboards.fullname" . }}-delivery
+  namespace: kyma-system
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/dashboards/templates/delivery.yaml]: Done

File: resources/eventing/charts/dashboards/templates/latency.yaml [hunk 9]
```diff
@@ -0,0 +1,371 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "dashboards.fullname" . }}-latency
+  namespace: kyma-system
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/dashboards/templates/latency.yaml]: Done

File: resources/eventing/charts/event-publisher-nats/templates/servicemonitor.yaml [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "event-publisher-nats.labels" . | nindent 4 }}
+    prometheus: monitoring
+spec:
+  selector:
+    matchLabels: {{- include "event-publisher-nats.labels" . | nindent 6 }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace | quote }}
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/event-publisher-nats/templates/servicemonitor.yaml]: Removed

File: resources/eventing/charts/event-publisher-nats/templates/service.yaml [hunk 11]
```diff
@@ -8,22 +8,24 @@ spec:
   type: ClusterIP
   selector: {{- include "event-publisher-nats.selectorLabels" . | nindent 4 }}
   ports:
-    - protocol: TCP
+    - name: {{ .Values.nats.config.portName }}
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/event-publisher-nats/templates/service.yaml:11]: Will be addressed here: https://github.com/kyma-project/kyma/issues/11113.

File: resources/eventing/charts/event-publisher-nats/templates/peerauthentication.yaml [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: {{ .Release.Namespace }}
+  labels: {{- include "event-publisher-nats.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels: {{- include "event-publisher-nats.selectorLabels" . | nindent 6 }}
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/event-publisher-nats/templates/peerauthentication.yaml:10]: Will be addressed here: https://github.com/kyma-project/kyma/issues/11113.

File: resources/eventing/charts/event-publisher-nats/templates/peerauthentication.yaml [hunk 13]
```diff
@@ -0,0 +1,14 @@
+{{- if eq .Values.global.eventing.backend "nats" }}
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/event-publisher-nats/templates/peerauthentication.yaml:1]: For now, we need this when NATS is the eventing backend.

File: resources/eventing/charts/nats-controller/templates/peerauthentication.yaml [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: {{ .Release.Namespace }}
+  labels: {{- include "nats-controller.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels: {{- include "nats-controller.selectorLabels" . | nindent 6 }}
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply resources/eventing/charts/nats-controller/templates/peerauthentication.yaml:10]: Will be addressed here: https://github.com/kyma-project/kyma/issues/11113.

**Review by sayanh — APPROVED** [review]

**Review by marcobebway — APPROVED** [review]

**Pull request merged**